azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/ConfigMojo.java

```diff
@@ -0,0 +1,342 @@
+        if (target == null) {
+            target = MavenUtils.createPlugin(this.plugin.getGroupId(), this.plugin.getArtifactId(), this.plugin.getVersion());
+            if (model.getBuild() == null) {
+                model.setBuild(new Build());
+            }
+            model.getBuild().addPlugin(target);
+        }
+        if (StringUtils.isBlank(target.getVersion())) {
+            target.setVersion(this.plugin.getVersion());
+        }
+
+        if (target.getConfiguration() == null) {
+            target.setConfiguration(new Xpp3Dom("configuration"));
+        }
+        final Xpp3Dom config = (Xpp3Dom) target.getConfiguration();
+
+        XmlUtils.replaceDomWithKeyValue(config, "subscriptionId", this.subscriptionId);
+        XmlUtils.replaceDomWithKeyValue(config, "clusterName", this.clusterName);
+        XmlUtils.replaceDomWithKeyValue(config, "appName", this.appName);
+        if (this.isPublic != null) {
+            XmlUtils.replaceDomWithKeyValue(config, "isPublic", this.isPublic);
+        }
+        if (includeDeployment && this.deployment != null) {
+            if (config.getChild(DEPLOYMENT_TAG) == null) {
+                config.addChild(XmlUtils.createDomWithName(DEPLOYMENT_TAG));
+            }
+            this.deployment.applyToXpp3Dom(config.getChild(DEPLOYMENT_TAG));
+        }
+
+        Writer writer = null;
+        try {
+            // this code is copied from https://github.com/Commonjava/maven-model-jdom-support
+            // which can apply the xml changes and preserve existing order of elements, formatting, and comments
+            final File pom = proj.getFile();
+            final SAXBuilder builder = new SAXBuilder();
+            builder.setIgnoringBoundaryWhitespace(false);
+            builder.setIgnoringElementContentWhitespace(false);
+            final Document doc = builder.build(pom);
+            String encoding = model.getModelEncoding();
+            if (encoding == null) {
+                encoding = "UTF-8";
+            }
+
+            final Format format = Format.getPrettyFormat().setEncoding(encoding).setTextMode(TextMode.PRESERVE);
+            writer = WriterFactory.newWriter(pom, encoding);
+            new MavenJDOMWriter().write(model, doc, writer, format);
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        } finally {
+            IOUtil.close(writer);
+        }
+    }
+
+    private void configurePublic() throws IOException {
+        this.isPublic = prompt.promoteYesNo(false, "Do you want to enable public access for this app?", true);
+    }
+
+    private void configurePublicParent() throws IOException {
+        publicProjects = prompt.promoteMultipleEntities("project", targetProjects, t -> t.getName(), true, " to select none",
+                new ArrayList<MavenProject>());
+    }
+
+    private void configureAppName() throws IOException {
+        appName = prompt.promoteString("app name", "${project.artifactId}", "[A-Za-z0-9_\\.]+", true);
```

Comment:
Will maven artifactId didn't follow [A-Za-z0-9_\\.]+?

Reply:
all the string values should have regex from PM, I am asking for the final one and will update it.